parlai/core/torch_agent.py

```diff
@@ -345,6 +415,10 @@ def reset(self):
         self.episode_done = True
 
     def observe(self, observation):
+        """Process incoming message in preparation for producing a response.
+
+        This includes remembering the past history of the conversation.
+        """
         observation = copy.deepcopy(observation)
```

Comment:
possibly slow?

Reply:
agreed, I want to replace this function with calling maintain_dialog_history too but happy to remove this in the meantime